### PR TITLE
add tasks and plans directories

### PR DESCRIPTION
**Pull request by ghost**
Hey folks,

Whenever any of my customers/prospects beginning using PE and the control repo, adding tasks and plans is always a bit of a mystery, especially when there's no template directories or files like there are for roles and profiles. 

Given the maturity of tasks and plans at this point, we should really include something in the template control repo.

The `adhoc` directory within `site-modules` contains a `tasks` and `plans` directory. 

In the tasks directory I've included a skeleton for a nix and windows based task with some basic task metadata (environment + task description). I've also included a plan with a description and targets param.

### CODE REVIEW

File: site-modules/adhoc/plans/myplan.pp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# This is a description for my plan
+plan adhoc::myplan(
+# input parameters go here
```
Comment (reidmv) [inline site-modules/adhoc/plans/myplan.pp]: Indent to level of line number (line 4) being described
```suggestion
  # input parameters go here
```

File: site-modules/adhoc/tasks/nix_example.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+     "description": "This is the description for the nix_example task",
+     "input_method": "environment"
+   }
```
Comment (reidmv) [inline site-modules/adhoc/tasks/nix_example.json]: Indentation for JSON: 4 spaces, closing brace at same level as opening brace
```suggestion
    "description": "This is the description for the nix_example task",
    "input_method": "environment"
}
```

File: site-modules/adhoc/plans/myplan.pp [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  TargetSpec $targets,
+) {
+
+# plan steps go here
```
Comment (reidmv) [inline site-modules/adhoc/plans/myplan.pp]: Comment at indentation level content would be. It's Puppet DSL, so two spaces in.
```suggestion
  # plan steps go here
```

File: site-modules/adhoc/tasks/win_example.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+     "description": "This is the description for the win_example task",
+     "input_method": "powershell"
+   }
```
Comment (reidmv) [inline site-modules/adhoc/tasks/win_example.json]: Indentation for JSON: 4 spaces, closing brace at same level as opening brace
```suggestion
    "description": "This is the description for the win_example task",
    "input_method": "powershell"
}
```

File: site-modules/adhoc/tasks/win_example.ps1 [hunk 5]
```diff
@@ -0,0 +1 @@
+# script content goes here
```
Comment (reidmv) [inline site-modules/adhoc/tasks/win_example.ps1]: In Bolt, scripts and tasks are different. Scripts go in `scripts/`. This is a task, so comment should use that language.
```suggestion
# task content goes here
```

File: site-modules/adhoc/tasks/nix_example.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+# script content goes here
```
Comment (reidmv) [inline site-modules/adhoc/tasks/nix_example.sh]: In Bolt, scripts and tasks are different. Scripts go in scripts/. This is a task, so comment should use that language.
```suggestion
# task content goes here
```

**Review by reidmv — CHANGES_REQUESTED** [review]
I like the idea of including a sample module for quick tasks/plans in a control repo.

I haven't thought a lot about the name of such a module, but looking at this suggestion of "adhoc", I'm a fan. I think it gets the idea across, and the literal definition of the term is perfect for control-repo content:

> In English, it generally signifies a solution designed for a specific problem or task, non-generalizable, and not intended to be adapted to other purposes

([Wikipedia](https://en.wikipedia.org/wiki/Ad_hoc)).

I've included a bunch of requested changes in this review, but they are just nit-level changes. Overall I'm 👍 to this. I'd support merging this in once those issues are fixed, and a second reviewer takes a look.

**Comment by ghost** [comment]
Awesome! Thanks for the suggestions, I've accepted them now - ready for second review.

**Review by reidmv — APPROVED** [review]
LGTM! 👍

**Comment by npwalker** [comment]
fwiw, bolt has `bolt plan new proj_name::plan_name` with the optional `-pp` flag.  You get a YAML plan without the `-pp` flag and you get a puppet language plan with the `-pp` flag.  I'd argue the plan examples should just be the same one from those commands.  

The long term goal is to be able to run projects directly in PE and forgo the need to put them in a control-repo or module but this seems fine in the interim.

**Pull request merged**